src/plugins/xerces/deserializer.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file
+ *
+ * @brief deserialization implementation for xerces plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include "deserializer.hpp"
+#include "util.hpp"
+
+#include <xercesc/dom/DOM.hpp>
+#include <xercesc/dom/DOMAttr.hpp>
+#include <xercesc/dom/DOMDocument.hpp>
+#include <xercesc/dom/DOMImplementation.hpp>
+#include <xercesc/dom/DOMNode.hpp>
+#include <xercesc/parsers/XercesDOMParser.hpp>
+
+#include <kdblogger.h>
+#include <key.hpp>
+
+XERCES_CPP_NAMESPACE_USE
+using namespace std;
+using namespace kdb;
+using namespace xerces;
+
+namespace
+{
+
+XercesPtr<DOMDocument> doc2dom (std::string const & src)
+{
+	XercesDOMParser parser;
+	parser.setValidationScheme (XercesDOMParser::Val_Auto);
+	parser.setDoNamespaces (false);
+	parser.setDoSchema (false);
+	parser.setCreateEntityReferenceNodes (false);
+
+	parser.parse (asXMLCh (src));
+	return XercesPtr<DOMDocument> (parser.adoptDocument ());
+}
+
+string getElementText (DOMNode const * parent)
+{
+	string str;
+
+	for (auto child = parent->getFirstChild (); child != NULL; child = child->getNextSibling ())
+	{
+		if (DOMNode::TEXT_NODE == child->getNodeType () || DOMNode::CDATA_SECTION_NODE == child->getNodeType ())
+		{
+			DOMText * data = dynamic_cast<DOMText *> (child);
+			if (!data->getIsElementContentWhitespace ()) str += toStr (data->getData ());
+		}
+	}
+	// trim whitespace
```

Comment:
> you never fail to point out erroneous parts :)

I wish I would see all of them, then we would have 200 issues less ;)

Thanks for the detailed analysis! Please document this in the README.md.
Ignoring whitespace-only-nodes looks like a good compromise, doesn't it?
It should cover the pretty-printing issues. Other spaces should be taken as-is, you never know if they are significant.